pkg/solana/txm/pendingtx.go

```diff
@@ -3,78 +3,122 @@ package txm
 import (
 	"context"
 	"errors"
+	"fmt"
 	"sync"
 	"time"
 
 	"github.com/gagliardetto/solana-go"
-	"github.com/google/uuid"
 	"golang.org/x/exp/maps"
 )
 
+var ErrAlreadyInExpectedState = errors.New("transaction already in expected state")
```

Comment:
Maybe add more granular erroring?
```suggestion
func ErrAlreadyInExpectedState(state TxState) error {
    return errors.New(fmt.Sprintf("transaction already in expected state: %s", state.String()))
}
```

Reply:
This was a good suggestion but think this is causing some problems with `errors.Is` for some reason. I think it's ok if we skip appending the state here. We use this predefined error to skip logging actually so the message itself probably isn't as important.

Reply:
Was curious about this. It seems the problem is the following:

- Every call to `ErrAlreadyInExpectedState(state)` creates a new error instance
- The `errors.Is` function checks if any error in the chain matches the target error instance, not the message. 
- Since each call generates a distinct error instance, `errors.Is(err, ErrAlreadyInExpectedState(state))` will always return false because they are different instances in memory.

Reply:
Right, I was thinking it was just a string match under the hood until running into this issue

Reply:
We could potentially do a string match for that check but it probably isn't necessary

Reply:
Right, we could but think it's better to skip since we don't even log this err on the TXM side. I actually added this predefined error to avoid noise in the logs.